src/index.ts

```diff
@@ -109,3 +109,49 @@ export const compare = memoize((versionA: string | null, versionB: string | null
 	}
 	return versionA.localeCompare(versionB);
 }, (a: string, b: string) => `${a} && ${b}`);
+
+/**
+ * @summary Return the major version number
+ * @name compare
+ * @public
+ * @function
+ *
+ * @description Returns the major version number in a semver string.
+ * If the presented version is a falsey value, it returns `0`. If the version is
+ * not a valid semver string, it returns the first number it finds in the string.
+ * If there are no numbers in the provided string, it returns `1`.
+ *
+ * @param {string|null} version - The version string to evaluate
+ *
+ * @returns {number} - The major version number
+ *
+ * @example
+ * resinSemver.major(null); //--> 0
+ *
+ * resinSemver.major('Resin OS v2.0.5'); //--> 2
+ *
+ * resinSemver.major('Resin OS v2.0.5'); //--> 2
+ *
+ * resinSemver.major('Linux 14.04'); //--> 14
+ *
+ * resinSemver.major('My development version'); //--> 1
```

Comment:
by semverish, I mean a string that is either a valid semver, or a semver string that has the resin prefixes and suffixes, EG "ResinOS 2.0.2+rev2 (prod)".